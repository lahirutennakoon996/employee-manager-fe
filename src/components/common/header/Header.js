import cx from 'classnames';

import styles from './Header.module.css';

export default function Header () {
  return (
    <nav className={cx("navbar", styles.navbar)}>
      <div className="container-fluid">
        <h3 className={styles.heading}>Employee Manager</h3>
      </div>
    </nav>
  )
}