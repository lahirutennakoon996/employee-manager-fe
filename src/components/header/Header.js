import cx from 'classnames';

import styles from './Header.module.css';

export default function Header () {
  return (
    <nav className={cx("navbar", styles.navbar)}>
      <div className="container-fluid">
        <a className="navbar-brand" href="#">Employee Manager</a>
      </div>
    </nav>
  )
}