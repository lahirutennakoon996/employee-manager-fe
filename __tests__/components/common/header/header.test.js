import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';

import Header from "@/components/common/header/Header";
import mockStore from "../../../../__mocks__/react-redux.mock";

describe('Header', () => {
  const initialState = {
    employees: {
      allEmployees: [],
      singleEmployee: null,
    },
  };

  const store = mockStore(initialState);
  let component;

  beforeEach(() => {
    const componentRendered = render(<Provider store={store}><Header /></Provider>);
    component = componentRendered.container;
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('render the heading', () => {
    const heading = screen.getByRole('heading', {
      name: /Employee Manager/i,
    });
    expect(heading).toBeInTheDocument();
  })
})