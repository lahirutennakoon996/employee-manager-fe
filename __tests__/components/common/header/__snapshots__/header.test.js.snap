// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div>
  <nav
    class="navbar navbar"
  >
    <div
      class="container-fluid"
    >
      <h3
        class="heading"
      >
        Employee Manager
      </h3>
    </div>
  </nav>
</div>
`;
